server/src/memory/prompts.rs

```diff
@@ -0,0 +1,164 @@
+use color_eyre::Result;
+use sqlx::PgPool;
+
+use super::blocks::MemoryBlock;
+
+#[derive(Debug)]
+pub struct PromptGenerator;
+
+impl PromptGenerator {
+    pub fn new() -> Self {
+        Self
+    }
+
+    pub async fn generate_system_prompt(pool: &PgPool, is_discord_context: bool) -> Result<String> {
```

Comment:
We are going to pass in a `&Thread` here instead of the `is_discord_context` and we will use the type of thread to determine if we want discord

We also want to add the thread goal to the system prompt so the agent knows what the goal for this current thread is